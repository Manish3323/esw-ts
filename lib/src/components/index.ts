import AuthContextProvider from './aas/context/AuthContextProvider'
import RealmRole from './aas/authorization/RealmRole'
import CheckLogin from './aas/authentication/CheckLogin'
import Login from './aas/Login'
import Logout from './aas/Logout'
import ClientRole from './aas/authorization/ClientRole'
import { Consumer, AuthContext } from './aas/context/AuthContext'

// #export-components
export {
  Logout,
  Login,
  CheckLogin,
  RealmRole,
  ClientRole,
  AuthContextProvider,
  Consumer,
  AuthContext
}
// #export-components
