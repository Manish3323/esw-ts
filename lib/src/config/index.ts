export { AASConfig } from './AASConfig'
export { LocationConfig } from './LocationConfig'
export {
  AGENT_SERVICE_CONNECTION,
  AUTH_CONNECTION,
  CONFIG_CONNECTION,
  GATEWAY_CONNECTION,
  SEQUENCE_MANAGER_CONNECTION,
  resolveConnection,
  eswAgentConnection
} from './Connections'
export { setAppConfigPath } from './AppConfigPath'
