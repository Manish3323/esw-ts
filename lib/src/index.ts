export * from './clients/command'
export * from './clients/location'
export * from './clients/sequencer'
export * from './clients/alarm'
export * from './clients/aas'
export * from './models'
export * from './config'
export * from './components'
