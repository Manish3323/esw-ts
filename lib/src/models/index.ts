export { ComponentId } from './ComponentId'
export type { ComponentType } from './ComponentType'
export { BaseKey } from './params/BaseKey'
export { ChoiceKeyFactory } from './params/ChoiceKeyFactory'
export { CurrentState } from './params/CurrentState'
export { Parameter } from './params/Parameter'
export { Prefix } from './params/Prefix'
export { Struct } from './params/Struct'
export type { Subsystem } from './params/Subsystem'
export type { Units } from './params/Units'
export { Result } from './params/Result'
export { Observe, SequenceCommand, ControlCommand, Setup, Wait } from './params/Command'
export type {
  OnewayResponse,
  SubmitResponse,
  Accepted,
  Cancelled,
  CancelledResponse,
  Completed,
  Error,
  ErrorResponse,
  InvalidResponse,
  CompletedResponse,
  Invalid,
  Locked,
  LockedResponse,
  Started,
  ValidateResponse,
  StartedResponse,
  CommandResponse
} from './params/CommandResponse'
export type {
  AltAzCoord,
  CometCoord,
  Coord,
  EqCoord,
  EqFrame,
  MinorPlanetCoord,
  RaDec,
  SolarSystemCoord,
  SolarSystemObject,
  ProperMotion,
  Tag
} from './params/Coord'

// export only types
export type {
  IntKey,
  LongKey,
  ShortKey,
  FloatKey,
  DoubleKey,
  ByteKey,
  StringKey,
  CharKey,
  StructKey,
  ChoiceKey,
  IntMatrixKey,
  ByteMatrixKey,
  LongMatrixKey,
  ShortMatrixKey,
  FloatMatrixKey,
  DoubleMatrixKey,
  IntArrayKey,
  ByteArrayKey,
  LongArrayKey,
  ShortArrayKey,
  FloatArrayKey,
  DoubleArrayKey,
  BooleanKey,
  UTCTimeKey,
  TAITimeKey,
  RaDecKey,
  EqCoordKey,
  SolarSystemCoordKey,
  MinorPlanetCoordKey,
  CometCoordKey,
  AltAzCoordKey,
  CoordKey,
  Key
} from './params/Key'

// helpers functions
export {
  intKey,
  longKey,
  shortKey,
  floatKey,
  doubleKey,
  byteKey,
  stringKey,
  charKey,
  booleanKey,
  byteMatrixKey,
  intMatrixKey,
  longMatrixKey,
  shortMatrixKey,
  floatMatrixKey,
  doubleMatrixKey,
  byteArrayKey,
  intArrayKey,
  longArrayKey,
  shortArrayKey,
  floatArrayKey,
  doubleArrayKey,
  structKey,
  utcTimeKey,
  taiTimeKey,
  choiceKey,
  raDecKey,
  eqCoordKey,
  solarSystemCoordKey,
  minorPlanetCoordKey,
  cometCoordKey,
  altAzCoordKey,
  coordKey
} from './params/Key'
