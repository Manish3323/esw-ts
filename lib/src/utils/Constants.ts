import type { AgentService } from '../clients/agent-service'
import type { AlarmService } from '../clients/alarm'
import type { CommandService } from '../clients/command'
import type { EventService } from '../clients/event'
import type { LocationService } from '../clients/location'
import type { SequenceManagerService } from '../clients/sequence-manager'
import type { AuthContextType } from '../components/aas/context/AuthContext'

export const APP_NAME = 'App-Name'

declare global {
  interface Window {
    isMocked?: boolean

    instance<T>(mockedValue: T): T

    mockedAlarmService: AlarmService
    mockedEventService: EventService
    mockedCommandService: CommandService
    mockedLocationService: LocationService
    mockedAgentService: AgentService
    mockedSequenceManager: SequenceManagerService
    auth: AuthContextType
  }
}
