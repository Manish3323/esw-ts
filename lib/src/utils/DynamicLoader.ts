export const dynamicImport = (path: string) => {
  return import(path)
}
