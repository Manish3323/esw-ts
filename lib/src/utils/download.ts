export const download = (object: any, fileName: string) => {
  const anchorElement = document.createElement('a')
  anchorElement.href = window.URL.createObjectURL(object)
  anchorElement.download = fileName
  document.body.appendChild(anchorElement)
  anchorElement.click()
  anchorElement.remove()
}
