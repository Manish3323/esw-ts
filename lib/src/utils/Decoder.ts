import { pipe } from 'fp-ts/lib/pipeable'
import * as D from 'io-ts/lib/Decoder'

export type Decoder<T> = D.Decoder<unknown, T>

export const char = pipe(
  D.string,
  D.refine((s): s is string => s.length == 1, 'single char')
)
