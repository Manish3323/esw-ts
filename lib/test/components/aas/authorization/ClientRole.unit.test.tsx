import React from 'react'
import { render } from '@testing-library/react'
import { mocked } from 'ts-jest/utils'
import Keycloak from 'keycloak-js'
import { mockAuth, mockedKeyCloakInstance } from '../../../helpers/MockHelpers'
import { AuthContext } from '../../../../src/components'
import ClientRole from '../../../../src/components/aas/authorization/ClientRole'

jest.mock('keycloak-js')
// DEOPSCSW-630 - Javascript adapter for AAS
// DEOPSCSW-636 - JS adapter support  for Authorization
describe('<ClientRole />', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const mockFn = mocked(Keycloak, true)
  const keycloakInstance = mockedKeyCloakInstance()
  mockFn.mockReturnValue(keycloakInstance)

  test('should render children elements if authentication is true and with valid Client role | ESW-330', () => {
    const props = {
      children: <div className="client-role">Authentication successful</div>,
      error: <div className="error">Authentication unsuccessful</div>,
      clientRole: 'test-client-role',
      client: 'test-client'
    }

    const { container } = render(
      <AuthContext.Provider value={{ auth: mockAuth(true), logout: jest.fn(), login: jest.fn() }}>
        <ClientRole {...props} />
      </AuthContext.Provider>
    )

    expect(container.getElementsByClassName('client-role').length).toBe(1)
    expect(container.getElementsByClassName('error').length).toBe(0)
    expect(container).toMatchSnapshot()
  })

  test('should not render children elements if authentication is true but invalid Client role | ESW-330', () => {
    const props = {
      children: <div className="client-role">Authentication successful</div>,
      error: <div className="error">Authentication unsuccessful</div>,
      clientRole: 'invalid-client-role',
      client: 'invalid-client'
    }

    const { container } = render(
      <AuthContext.Provider
        value={{
          auth: mockAuth(true, false, 'test-realm-roles'),
          logout: jest.fn(),
          login: jest.fn()
        }}
      >
        <ClientRole {...props} />
      </AuthContext.Provider>
    )

    expect(container.getElementsByClassName('client-role').length).toBe(0)
    expect(container.getElementsByClassName('error').length).toBe(1)
    expect(container).toMatchSnapshot()
  })

  test('should not render children elements if authentication is false | ESW-330', () => {
    const props = {
      children: <div className="client-role">Authentication successful</div>,
      error: <div className="error">Authentication unsuccessful</div>,
      clientRole: 'invalid-client-role',
      client: 'invalid-client'
    }

    const { container } = render(
      <AuthContext.Provider
        value={{
          auth: mockAuth(true, false, 'test-realm-roles'),
          logout: jest.fn(),
          login: jest.fn()
        }}
      >
        <ClientRole {...props} />
      </AuthContext.Provider>
    )

    expect(container.getElementsByClassName('client-role').length).toBe(0)
    expect(container.getElementsByClassName('error').length).toBe(1)
    expect(container).toMatchSnapshot()
  })
})
