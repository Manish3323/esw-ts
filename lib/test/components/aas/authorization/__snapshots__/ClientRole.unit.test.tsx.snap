// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientRole /> should not render children elements if authentication is false  1`] = `
<div>
  <div>
    <div
      class="error"
    >
      Authentication unsuccessful
    </div>
  </div>
</div>
`;

exports[`<ClientRole /> should not render children elements if authentication is true but invalid Client role 1`] = `
<div>
  <div>
    <div
      class="error"
    >
      Authentication unsuccessful
    </div>
  </div>
</div>
`;

exports[`<ClientRole /> should render children elements if authentication is true and with valid Client role 1`] = `
<div>
  <div>
    <div
      class="client-role"
    >
      Authentication successful
    </div>
  </div>
</div>
`;
